thrift/server.go

```diff
@@ -185,6 +186,17 @@ func (s *Server) handle(origCtx context.Context, handler handler, method string,
 		return err
 	}
 
+	// Buffer the response write so we can send a system error to the client.
+	// It's too late to send a system error by the time we write Arg3.
+	var respStruct bytes.Buffer // XXX should we make a buffer pool?
```

Comment:
That’s a decision best made with data, but I expect pooling to yield a 20% improvement in latency.